File: node-sdk/manual_mode.mjs
/**
 * RoutePlex Node.js SDK — Manual Model Selection
 *
 * npm install @routeplex/node
 */

import { RoutePlex } from "@routeplex/node";

const client = new RoutePlex({ apiKey: process.env.ROUTEPLEX_API_KEY });

const response = await client.chat("Write a function to check if a number is prime", {
  model: "gpt-4o-mini",
  maxOutputTokens: 1024,
  temperature: 0.3,
});

console.log(response.output);
console.log(`Model: ${response.modelUsed}`);
console.log(`Tokens: ${response.usage.totalTokens}`);

File: python-sdk/manual_mode.py
"""
RoutePlex Python SDK — Manual Model Selection

pip install routeplex
"""

import os
from routeplex import RoutePlex

client = RoutePlex(api_key=os.environ["ROUTEPLEX_API_KEY"])

# Pick a specific model
response = client.chat(
    "Write a Python function to check if a number is prime",
    model="gpt-4o-mini",
    max_output_tokens=1024,
    temperature=0.3,
)

print(response.output)
print(f"Model: {response.model_used}")
print(f"Tokens: {response.usage.total_tokens}")
